editor/sidebar/post-visibility/index.js

```diff
@@ -111,12 +129,17 @@ class PostVisibility extends Component {
 					aria-expanded={ this.state.opened }
 					className="editor-post-visibility__toggle button-link"
 					onClick={ this.toggleDialog }
+					ref={ toggle => this.toggle = toggle }
 				>
 					{ getVisibilityLabel( visibility ) }
 				</button>
 
 				{ this.state.opened &&
-					<div className="editor-post-visibility__dialog">
+					<div className="editor-post-visibility__dialog"
+						onKeyDown={ this.handleKeyDown }
```

Comment:
If the rule causes more harm or confusion than it helps, I'd also be okay with disabling it globally from the root [`.eslintrc.json`](https://github.com/WordPress/gutenberg/blob/master/.eslintrc.json) (i.e. `"jsx-a11y/no-static-element-interactions": "off"`).